hogben/optimise.py

```diff
@@ -60,6 +61,9 @@ def optimise_angle_times(
         # Check that the measurement angle of the sample can be varied.
         assert isinstance(self.sample, VariableAngle)
 
+        # Set contrasts to empty list if not provided
+        contrasts = contrasts or []
```

Comment:
Is it not better to set the default in the init to `contrasts: Optional[list] = [],` this evaluates to `None` if required and removes this line